pkg/cloudprovider/cloud.go

```diff
@@ -37,6 +37,8 @@ type Interface interface {
 	Clusters() (Clusters, bool)
 	// Routes returns a routes interface along with whether the interface is supported.
 	Routes() (Routes, bool)
+	// Volumes returns a volumes interface along with whether the interface is supported.
+	Volumes() (Volumes, bool)
```

Comment:
@thockin But only 1 of those volume types have an API that can be used for provisioning.

NFS/iSCSI/Gluster/Ceph/etc. would all be "statically provisioned" and the current recycler can works for those (does for NFS already).

GCEPD/EBS/Cinder are the only 3 we have with an API.  Those APIs seem to have just 1 kind of volume in the API, but many types you can request via various performance attributes.